aspnetcore/signalr/Configuration.md

```diff
@@ -0,0 +1,124 @@
+
+The `HubOptions` are configured in the `ConfigureServices` method of the `Startup` class, as shown in the following code sample:
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddSignalR(hubOptions =>
+    {
+        hubOptions.EnableDetailedErrors = true;
+        hubOptions.KeepAliveInterval = TimeSpan.FromMinutes(1);
+    })
+}
+```
+
+Strongly-typed [HubOptions<T>](https://github.com/aspnet/SignalR/blob/dev/src/Microsoft.AspNetCore.SignalR.Core/HubOptions.cs) can be configured as follows: 
+
+```csharp
+  services.AddSignalR().AddHubOptions<HubName>(options =>
+  {
+      options.EnableDetailedErrors = true;
+  }
+```
+
+### HttpConnectionDispatcherOptions
+
+The following table describes the `HttpConnectionDispatcherOptions` options:
+
+| Option | Description |
+| ------ | ----------- |
+| `ApplicationMaxBufferSize`  | The maximum number of bytes the connection (transport) can buffer when invoking methods from the client, before blocking and waiting for the application to consume enough bytes to allow writing again.  |
+| `AuthorizationData` | A pre-populated list of `IAuthorizeData` gathered from the `Authorize` attributes used on the `Hub<T>`. |
+| `LongPolling`  | Gets the LongPolling options object that has a settable `PollTimeout` property. Setting the `PollTimeOut` sets the wait time before ending a poll request. |
+| `TransportMaxBufferSize`  | The maximum number of bytes the application. For example, `Clients.All.SendAsync` can buffer when writing to a single connection before blocking and waiting for the connection to consume enough bytes to allow writing again. |
+| `Transports`  | A bitmask that sets a transport. The available `options.Transports` options are as follows: `HttpTransportType.WebSockets`, `HttpTransportType.LongPolling`, and `HttpTransportType.ServerSentEvents`
+    WebSockets - Gets the WebSockets options object.  |
+| `WebSockets`  |  Gets the WebSockets options object.  |
+| `CloseTimeout`  | TimeSpan to set how long to wait for a clean WebSocket close when connection is being terminated  |
+| `SubProtocolSelector` | A delegate the hub calls and passes a list of `Sec-WebSocket-Protocol` values from the request header. A delegate returns the chosen `Sec-WebSocket-Protocol`. |
+
+The following code samples demonstrate setting the connection options:
+
+```csharp
+var connection = new HubConnectionBuilder()
+    .WithUrl("...", HttpTransportType.WebSockets | HttpTransportType.ServerSentEvents)
+    .Build();
+
+var connection = new HubConnectionBuilder()
+    .WithUrl("...", options => {
+        options.AccessTokenProvider = async () => {
+            // Get access token and return it.
+        };
+    });
+    .Build();
+
+var connection = new HubConnectionBuilder()
+    .WithUrl("...", options => {
+        options.Headers["Foo"] = "Bar";
+        options.Cookies.Add(new Cookie(...));
+        options.ClientCertificates.Add(...);
```

Comment:
For the table

Headers: HTTP headers to be applied to all HTTP requests.
Cookies: Cookies to be sent with each HTTP request.
ClientCertificates: TLS Client Certificates to send when connecting over HTTP (**not supported in Xamarin**)

Reply:
Note that headers, cookies and client certificates **cannot be configured in the JavaScript client** due to limitations on browser APIs. We should put a note in the section documenting JavaScript client options to avoid confusion.